tracer/src/Datadog.Trace/Iast/StackWalker.cs

```diff
@@ -55,7 +55,8 @@ public static StackTrace GetStackTrace()
     public static bool TryGetFrame(StackTrace stackTrace, out StackFrame? targetFrame)
     {
         targetFrame = null;
-        foreach (var frame in stackTrace.GetFrames())
+        var frames = stackTrace.GetFrames() ?? Enumerable.Empty<StackFrame>();
```

Comment:
Or maybe bail out if it's null

Reply:
If frames is null/empty, we would be returning true with a null targetFrame. Then, we would create  a Location with null frame values. Is that what we want or would it be better to return false and add a null location to the vulnerability?

Reply:
I updated to return false because it seems more logical to return false in that case.
I let this PR in draft, waiting for an answer

Reply:
IMHO, it's fine to return false

Reply:
IIRC returning false avoided the creation of the vuln and it's only used for when we find a "specially excluded" frame in the stack. So I'd return true with null. We'd have to check if we support having null in the location, which I suspect we do

Reply:
Yes, we do
`
 public Location(StackFrame? stackFrame, StackTrace? stack, string? stackId, ulong? spanId)
    {
        var method = stackFrame?.GetMethod();
        Path = method?.DeclaringType?.FullName;
        Method = method?.Name;
        var line = stackFrame?.GetFileLineNumber();
        Line = line > 0 ? line : null;

        SpanId = spanId == 0 ? null : spanId;

        _stack = stack;
        StackId = stackId;
    }
`